comp/core/workloadmeta/collectors/internal/ecsfargate/ecsfargate.go

```diff
@@ -28,20 +32,26 @@ const (
 )
 
 type collector struct {
-	id      string
-	store   workloadmeta.Component
-	catalog workloadmeta.AgentType
-	metaV2  v2.Client
-	seen    map[workloadmeta.EntityID]struct{}
+	id            string
+	store         workloadmeta.Component
+	catalog       workloadmeta.AgentType
+	metaV2        v2.Client
+	metaV4        v3or4.Client
+	seen          map[workloadmeta.EntityID]struct{}
+	v4TaskEnabled bool
 }
 
 // NewCollector returns a new ecsfargate collector provider and an error
 func NewCollector() (workloadmeta.CollectorProvider, error) {
+	v4TaskEnabled := config.Datadog.GetBool("orchestrator_explorer.enabled") &&
+		config.Datadog.GetBool("orchestrator_explorer.ecs_collection.enabled") &&
+		(flavor.GetFlavor() == flavor.DefaultAgent)
```

Comment:
Would it be interesting to have this extracted into a function?

Reply:
I move this part to util pkg as it's shared with ecs and ecsfargate collector https://github.com/DataDog/datadog-agent/blob/882d6e4354906f23c0124aa0d908eed0e5bcbd4a/comp/core/workloadmeta/collectors/util/ecs_util.go#L23-L28